sound/soc/sof/topology.c

```diff
@@ -3330,12 +3329,9 @@ int snd_sof_complete_pipeline(struct snd_sof_dev *sdev,
 	ready.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_PIPE_COMPLETE;
 	ready.comp_id = swidget->comp_id;
 
-	ret = sof_ipc_tx_message(sdev->ipc,
-				 ready.hdr.cmd, &ready, sizeof(ready), &reply,
-				 sizeof(reply));
-	if (ret < 0)
-		return ret;
-	return 1;
+	return !sof_ipc_tx_message(sdev->ipc,
+				   ready.hdr.cmd, &ready, sizeof(ready), &reply,
+				   sizeof(reply));
```

Comment:
if res can be >0 then this is wrong.

Reply:
@lyakh the logic for this is completely broken I agree. I think we should just remove swidget->complete. It is unlikely that the widget is added to the list more than once.

If this logic did work, we'd never be able to do complete_pipeline after a suspend/resume cycle. So thank god for that :)